lib/postgrex/extensions/path.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Postgrex.Extensions.Path do
+  @moduledoc false
+  import Postgrex.BinaryUtils, warn: false
+  use Postgrex.BinaryExtension, send: "path_send"
+  alias Postgrex.Extensions.Path
+  alias Postgrex.Extensions.Point
+
+  def encode(_) do
+    quote location: :keep do
+      %Postgrex.Path{open: o, points: ps} when is_list(ps) and is_boolean(o) ->
+        open_byte = Path.open_to_byte(o)
+        len = <<length(ps)::int32>>
+        encoded_points = Enum.map(ps, &Point.encode_point(&1, Postgrex.Path))
+
+        # 1 byte for open/closed flag, 4 for length, 16 for each point
+        nbytes = 5 + 16 * length(ps)
+        [<<nbytes::int32>>, open_byte, len, encoded_points]
+      other ->
+        raise ArgumentError, Postgrex.Utils.encode_msg(other, Postgrex.Path)
+    end
+  end
+      
+  def decode(_) do
+    quote location: :keep do
+      <<nbytes::int32, path_data::binary-size(nbytes)>> ->
+        Path.decode_path(path_data)
+    end
+  end
+
+  def decode_path(<<o::int8, n::int32, point_data::binary-size(n)-unit(128)>>) do
+    open = (o == 1)
+    points = decode_points(n, point_data, [])
+    %Postgrex.Path{open: open, points: points}
+  end
+
+  def open_to_byte(true), do: <<1 :: int8>>
+  def open_to_byte(false), do: <<0 :: int8>>
+
+  defp decode_points(0, _, points), do: Enum.reverse(points)
```

Comment:
Please use binary match as first argument in recursive function as compiler can optimise. See http://erlang.org/doc/efficiency_guide/binaryhandling.html for more info. Also no need to count as checked length in initial match.